### ISSUE TITLE
bug: Patched APK does not install, crashes when exporting

### ISSUE THREAD

**Issue by CovalenceSTU** [kitadai31/revanced-manager-android5-7#7]
### Type

Other

### Bug description

When I create the patched Youtube APK (using the patches from this repo), it patches successfully but nothing happens when I install and when I try to export the APK, the app makes a 0 byte file and crashes.

### Steps to reproduce

1. Select the Youtube APK from storage (v17.34.36, from APKMirror)
2. Select some patches and patch it (I've tried as many as 22 and as few as 5, all with the same results)
3. Press Install after it is done, nothing will happen.
4. Press "Export APK" from the menu - when saving, a 0 byte file will be created then the app crashes a second later.

### Android version

7.0

### Manager version

1.6.1-API23

### Target app name

YouTube

### Target app version.

17.34.36

### Installation type

Non-root

### Patches selected.

enable-minimized-playback, hide-video-ads, microg-support, return-youtube-dislike, settings, sponsorblock

### Device logs (exported using Manager settings).

```shell
--------- beginning of crash
08-18 15:24:57.840  3964  3964 E AndroidRuntime: FATAL EXCEPTION: main
08-18 15:24:57.840  3964  3964 E AndroidRuntime: Process: app.revanced.manager.flutter, PID: 3964
08-18 15:24:57.840  3964  3964 E AndroidRuntime: java.io.FileNotFoundException: /data/user/0/app.revanced.manager.flutter/cache/patcher/tmp-CPRKNQ/out.apk (No such file or directory)
08-18 15:24:57.840  3964  3964 E AndroidRuntime: 	at java.io.FileInputStream.open(Native Method)
08-18 15:24:57.840  3964  3964 E AndroidRuntime: 	at java.io.FileInputStream.<init>(FileInputStream.java:146)
08-18 15:24:57.840  3964  3964 E AndroidRuntime: 	at com.cleveroad.cr_file_saver.utils.FileKt$saveFileInBackground$1$file$1.invokeSuspend(File.kt:71)
08-18 15:24:57.840  3964  3964 E AndroidRuntime: 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
08-18 15:24:57.840  3964  3964 E AndroidRuntime: 	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
08-18 15:24:57.840  3964  3964 E AndroidRuntime: 	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)
08-18 15:24:57.840  3964  3964 E AndroidRuntime: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
08-18 15:24:57.840  3964  3964 E AndroidRuntime: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)
08-18 15:24:57.840  3964  3964 E AndroidRuntime: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)
08-18 15:33:47.045 10101 10101 E AndroidRuntime: FATAL EXCEPTION: main
08-18 15:33:47.045 10101 10101 E AndroidRuntime: Process: app.revanced.manager.flutter, PID: 10101
08-18 15:33:47.045 10101 10101 E AndroidRuntime: java.io.FileNotFoundException: /data/user/0/app.revanced.manager.flutter/cache/patcher/tmp-CPBFQV/out.apk (No such file or directory)
08-18 15:33:47.045 10101 10101 E AndroidRuntime: 	at java.io.FileInputStream.open(Native Method)
08-18 15:33:47.045 10101 10101 E AndroidRuntime: 	at java.io.FileInputStream.<init>(FileInputStream.java:146)
08-18 15:33:47.045 10101 10101 E AndroidRuntime: 	at com.cleveroad.cr_file_saver.utils.FileKt$saveFileInBackground$1$file$1.invokeSuspend(File.kt:71)
08-18 15:33:47.045 10101 10101 E AndroidRuntime: 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
08-18 15:33:47.045 10101 10101 E AndroidRuntime: 	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
08-18 15:33:47.045 10101 10101 E AndroidRuntime: 	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)
08-18 15:33:47.045 10101 10101 E AndroidRuntime: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
08-18 15:33:47.045 10101 10101 E AndroidRuntime: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)
08-18 15:33:47.045 10101 10101 E AndroidRuntime: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)
08-18 15:58:11.057 10942 10942 E AndroidRuntime: FATAL EXCEPTION: main
08-18 15:58:11.057 10942 10942 E AndroidRuntime: Process: app.revanced.manager.flutter, PID: 10942
08-18 15:58:11.057 10942 10942 E AndroidRuntime: java.io.FileNotFoundException: /data/user/0/app.revanced.manager.flutter/cache/patcher/tmp-DPSPZG/out.apk (No such file or directory)
08-18 15:58:11.057 10942 10942 E AndroidRuntime: 	at java.io.FileInputStream.open(Native Method)
08-18 15:58:11.057 10942 10942 E AndroidRuntime: 	at java.io.FileInputStream.<init>(FileInputStream.java:146)
08-18 15:58:11.057 10942 10942 E AndroidRuntime: 	at com.cleveroad.cr_file_saver.utils.FileKt$saveFileInBackground$1$file$1.invokeSuspend(File.kt:71)
08-18 15:58:11.057 10942 10942 E AndroidRuntime: 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
08-18 15:58:11.057 10942 10942 E AndroidRuntime: 	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
08-18 15:58:11.057 10942 10942 E AndroidRuntime: 	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)
08-18 15:58:11.057 10942 10942 E AndroidRuntime: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
08-18 15:58:11.057 10942 10942 E AndroidRuntime: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)
08-18 15:58:11.057 10942 10942 E AndroidRuntime: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)
08-18 16:49:24.698 19629 19629 E AndroidRuntime: FATAL EXCEPTION: main
08-18 16:49:24.698 19629 19629 E AndroidRuntime: Process: app.revanced.manager.flutter, PID: 19629
08-18 16:49:24.698 19629 19629 E AndroidRuntime: java.io.FileNotFoundException: /data/user/0/app.revanced.manager.flutter/cache/patcher/tmp-TXUQXL/out.apk (No such file or directory)
08-18 16:49:24.698 19629 19629 E AndroidRuntime: 	at java.io.FileInputStream.open(Native Method)
08-18 16:49:24.698 19629 19629 E AndroidRuntime: 	at java.io.FileInputStream.<init>(FileInputStream.java:146)
08-18 16:49:24.698 19629 19629 E AndroidRuntime: 	at com.cleveroad.cr_file_saver.utils.FileKt$saveFileInBackground$1$file$1.invokeSuspend(File.kt:71)
08-18 16:49:24.698 19629 19629 E AndroidRuntime: 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
08-18 16:49:24.698 19629 19629 E AndroidRuntime: 	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
08-18 16:49:24.698 19629 19629 E AndroidRuntime: 	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)
08-18 16:49:24.698 19629 19629 E AndroidRuntime: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
08-18 16:49:24.698 19629 19629 E AndroidRuntime: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)
08-18 16:49:24.698 19629 19629 E AndroidRuntime: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)
08-18 17:00:23.768 21999 21999 E AndroidRuntime: FATAL EXCEPTION: main
08-18 17:00:23.768 21999 21999 E AndroidRuntime: Process: app.revanced.manager.flutter, PID: 21999
08-18 17:00:23.768 21999 21999 E AndroidRuntime: java.io.FileNotFoundException: /data/user/0/app.revanced.manager.flutter/cache/patcher/tmp-UKIMPV/out.apk (No such file or directory)
08-18 17:00:23.768 21999 21999 E AndroidRuntime: 	at java.io.FileInputStream.open(Native Method)
08-18 17:00:23.768 21999 21999 E AndroidRuntime: 	at java.io.FileInputStream.<init>(FileInputStream.java:146)
08-18 17:00:23.768 21999 21999 E AndroidRuntime: 	at com.cleveroad.cr_file_saver.utils.FileKt$saveFileInBackground$1$file$1.invokeSuspend(File.kt:71)
08-18 17:00:23.768 21999 21999 E AndroidRuntime: 	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
08-18 17:00:23.768 21999 21999 E AndroidRuntime: 	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
08-18 17:00:23.768 21999 21999 E AndroidRuntime: 	at kotlinx.coroutines.scheduling.CoroutineScheduler.runSafely(CoroutineScheduler.kt:571)
08-18 17:00:23.768 21999 21999 E AndroidRuntime: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.executeTask(CoroutineScheduler.kt:750)
08-18 17:00:23.768 21999 21999 E AndroidRuntime: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.runWorker(CoroutineScheduler.kt:678)
08-18 17:00:23.768 21999 21999 E AndroidRuntime: 	at kotlinx.coroutines.scheduling.CoroutineScheduler$Worker.run(CoroutineScheduler.kt:665)
--------- beginning of main
```

### Installer logs (exported using Installer menu option) [unneeded if issue is not during patching].

```shell
Initializing installer
Creating working directory
Copying original apk
Unpacking input apk
Reading dex files
Decoding AndroidManifest.xml only, because resources are not needed
Merging integrations
Deleting existing resource cache directory
Decoding resources
Executing patches
Applied enable-minimized-playback
Applied hide-video-ads
Applied microg-support
Applied return-youtube-dislike
Applied settings
Applied sponsorblock
Repacking patched apk
Compiling resources
Writing modified dex files
Finished!
Installing patched file using nonroot method
```

### Additional context

- My device is an (unrooted) Galaxy Tab S2 9.7 with a Snapdragon 652.
- Is there a high minimum amount of space needed? I had around 800-900MB free but didn't get any warnings during the patching.
- It shouldn't cause this issue, but the latest version for my device in the Play Store is v17.34.35 not v17.34.36. The patches cause errors on 17.34.35 anyways, so I didn't try it.

### Acknowledgements

- [X] I have searched the existing issues and this is a new and no duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I filled out all of the requested information in this issue properly.
- [X] The issue is related solely to the ReVanced Manager and not related to patching errors or patches
- [X] I have written the title and contents in English.

**Comment by kitadai31** [comment]
I guess it's storage space issue.
Patching apps needs more than 300MB space.
Android may have deleted the ReVanced Manager cache while patching due to low storage space.

**Comment by kitadai31** [comment]
A solution found. Remove keystore.
> Any keystore except standard manager keystore causing manager to crash https://github.com/ReVanced/revanced-manager/issues/1253

**Comment by kitadai31** [comment]
Fixed in v1.17.1
